stream-legal-entity/legal-entity-core/src/main/java/com/backbase/stream/LegalEntitySaga.java

```diff
@@ -371,14 +377,26 @@ private ProductGroupTask createProductGroupTask(LegalEntityTask streamTask, Prod
         return jobProfileUsers
             .flatMap(jobProfileUser -> upsertUser(streamTask, jobProfileUser.getUser())
                 .map(user -> {
+                    Mono<UserProfile> userProfile = upsertUserProfile(user);
+                    user.setUserProfile(userProfile);
```

Comment:
@bartveenstra @uesleilima @moondowner need a bit of help here. How should this be done not using block?